jamaljsr/polar
# Add support for Eclair nodes

### Description

This PR adds support for Eclair v0.3.3. These nodes have all of the same functionality as LND and c-lightning nodes, to deposit funds, open channels, make payments, etc. 

There is only one quirk which changes the previous default behavior for all networks. Eclair will not boot up if it connects to a Bitcoin node with 0 mined blocks. You have to mine at least one block for it to not immediately exit on startup. So Polar will now automatically mine the first block for all networks.

### ToDo
- [ ] Verify everything works on Windows & Linux

### Steps to Test

1. Create a new Network with an Eclair node
1. Start the network and confirm the Eclair node starts successfully
1. Deposit funds
1. Open a channel with another node
1. Make a payment to and from this node
1. add another Eclair node from the sidebar
1. Open the terminal and execute `eclair-cli getinfo` commands
1. View the node logs

### Screenshots

![image](https://user-images.githubusercontent.com/1356600/76675882-93f39200-6594-11ea-91cb-a7325a60710c.png)

<details>
<summary>View more screenshots</summary>

![image](https://user-images.githubusercontent.com/1356600/76675950-488db380-6595-11ea-9022-5dd802391335.png)

![image](https://user-images.githubusercontent.com/1356600/76675886-9fdf5400-6594-11ea-877f-e09878d28883.png)

![image](https://user-images.githubusercontent.com/1356600/76675899-b84f6e80-6594-11ea-8326-9ddd45b6effe.png)
</details>
